### ISSUE TITLE
Some track buttons should be grayed out when building on a 25-degree horizontal angle

### ISSUE THREAD

**Issue by ghost** [OpenRCT2/OpenRCT2#2192]
This bug is also in the original RCT2.

When building at a 45-degree angle, the banking icons are clickable even when the track is sloped, although clicking them doesn't really do anything. The track will still be without a bank. I don't believe there is a single coaster type in the game that can bank at this angle.
![rct2_banking_bug](https://cloud.githubusercontent.com/assets/15527244/10917775/1e66ccf4-8228-11e5-8fee-4cd3db45e42c.PNG)

**Comment by Patrik356b** [comment]
There is a few coasters that can do 45 deg with banking if i recall correctly.

**Comment by osconnoisseur** [comment]
Let us know if you find out which ones. I've never seen it before.

There's another similar 45-degree issue. Some coasters (Giga Coaster and Steel Twister Coaster are two that are affected) allow the flat to steep slope or steep slope to flat to be selected, even though these coasters can't actually do that.
![rct2_slope_bug](https://cloud.githubusercontent.com/assets/15527244/10928725/faf23288-8270-11e5-94f9-abd6a1d0a830.png)

**Comment by Patrik356b** [comment]
![http://puu.sh/l8Zw9/40f40fd996.png](rides)

**Comment by Patrik356b** [comment]
I think there might be a few more, didn't have the time to look further

**Comment by jensj12** [comment]
@Patrik356b That's not 45 deg in that image.
What it should do when you click that button (in OP) is go to flat track and add banking so you can build a flat banked turn afterwards. Or maybe you had to click flat track first.

**Comment by osconnoisseur** [comment]
No, that's not what I meant. Perhaps my terminology is confusing. I will use the word diagonal instead. By diagonal, I mean the direction you're going after placing one large curve track piece. I don't know of any coasters in the game can bank on a slope while diagonal. They can bank while diagonal when flat, but not on a slope.

**Comment by Patrik356b** [comment]
I misread the issue the 2nd time I looked at it.

**Comment by 733737** [comment]
new pieces have been added to the game for this case.

**Comment by Gymnasiast** [comment]
Checked and both issues are fixed as of v0.5.4.
